src/Umbraco.Web.UI.Client/src/common/directives/components/content/edit.controller.js

```diff
@@ -331,8 +331,35 @@
 
         // This is a helper method to reduce the amount of code repitition for actions: Save, Publish, SendToPublish
         function performSave(args) {
-
-
+            
+            // Check that all variants for publishing have a name.
+            if (args.action === "publish" || args.action === "sendToPublish") {
+                
+                if ($scope.content.variants) {
+                    var iVariant;
+                    for (var i = 0; i < $scope.content.variants.length; i++) {
+                        iVariant = $scope.content.variants[i];
+                        
+                        iVariant.notifications = [];// maybe not needed, need to investigate.
+                        
+                        if(iVariant.publish === true) {
+                            if (iVariant.name == null) {
+                                
+                                var tokens = [iVariant.language.name];
+                                
+                                localizationService.localize("publish_contentPublishedFailedByMissingName", tokens).then(function (value) {
+                                    iVariant.notifications.push({"message": value, "type": 1});
+                                });
+                                
+                                return $q.reject();
```

Comment:
@nielslyngsoe 

This reject would need to go inside the localize call which is itself a promise so you would need to:

```js
return localizationService.localize("publish_contentPublishedFailedByMissingName", tokens).then(function (value) {
                                    iVariant.notifications.push({"message": value, "type": 1});
                                    
                                     return $q.reject();
                                });

```

I'll fix up in my branch